homeassistant/components/intellifire/config_flow.py

```diff
@@ -71,31 +78,101 @@ async def _find_fireplaces(self):
         LOGGER.debug("Configured Hosts: %s", configured_hosts)
         LOGGER.debug("Not Configured Hosts: %s", self._not_configured_hosts)
 
-    async def _async_validate_and_create_entry(self, host: str) -> FlowResult:
-        """Validate and create the entry."""
+    async def validate_api_access_and_create_or_update(
+        self, *, host: str, username: str, password: str, serial: str
+    ):
+        """Validate username/password against api."""
+        ift_control = IntellifireControlAsync(fireplace_ip=host)
+
+        LOGGER.info("Attempting login with: %s %s", username, password)
```

Comment:
Don't log passwords.

Reply:
Also also let's log it as debug, it's already given back to the user if it fails
```suggestion
        LOGGER.debug("Attempting login with: %s %s", username, password)
```